src/domains/tae-app/00_monitor_incident.tf

```diff
@@ -143,9 +143,86 @@ resource "azurerm_monitor_scheduled_query_rules_alert_v2" "cstar-external-access
     }
   }
 
-
   tags = {
     key = "Incident Alert"
   }
 }
 
+resource "azurerm_monitor_scheduled_query_rules_alert_v2" "cstar-decrypting-problems" {
+
+  count = var.env_short == "p" ? 1 : 0
+
+  name                = "tae-${var.env_short}-decrypting-problems"
+  resource_group_name = data.azurerm_resource_group.monitor_rg.name
+  location            = data.azurerm_resource_group.monitor_rg.location
+
+  evaluation_frequency      = "P1D"
+  window_duration           = "P1D"
+  query_time_range_override = "P2D"
+  scopes                    = [data.azurerm_log_analytics_workspace.log_analytics.id]
+  severity                  = 0
+
+  auto_mitigation_enabled          = false
+  workspace_alerts_storage_enabled = false
+  description                      = "Triggers whenever less of 50% of sender's files aren't derypted for at least 5 senders."
+  display_name                     = "tae-${var.env_short}-decrypting-problems#INC"
+  enabled                          = true
+
+  skip_query_validation = false
+
+  action {
+    action_groups = [
+      data.azurerm_monitor_action_group.slack.id # cstar-status
+    ]
+  }
+
+  criteria {
+    metric_measure_column   = "AffectedSenders"
+    time_aggregation_method = "Total"
+    threshold               = 5
+    operator                = "GreaterThanOrEqual"
+    query                   = <<-QUERY
+      let regexExtractSenderCode = @"AGGADE\.([^\.]*)";
+      let decryptedFile = StorageBlobLogs
+          | where AccountName == 'cstarpblobstorage'
+          | where OperationName in ('PutBlob', 'PutBlock')
+              and StatusCode == 201
+              and Uri has "ade-transactions-decrypted"
+          | extend Filename = tostring(extract(@"ade-transactions-decrypted\/([^?]*)", 1, Uri))
+          | extend CompositionPre = substring(Filename,7,28)
+          | extend SenderCode = tostring(split(Filename,".")[1])
+          | distinct CompositionPre, SenderCode
+          | summarize CountPreDecryption = count() by CompositionPre, SenderCode;
+      StorageBlobLogs
+          | where AccountName == 'cstarpblobstorage'
+          | where OperationName in ('PutBlob', 'PutBlock')
+              and StatusCode == 201
+              and Uri has ".pgp"
+              and TimeGenerated >= ago(1d) and TimeGenerated < ago(30m)
```

Comment:
The logic is that it shoud be triggered every day, once a day, and it will be evaluated since 1 day ago to 30 minutes ago. Do you think that the window duration and the TimeGenerated's constraints can be collide?!